src/lib/Router/AppRouter.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import {
+  BrowserRouter as Router,
+  Route,
+  Routes
+} from 'react-router-dom';
+import { Link as RouterLink, LinkProps } from 'react-router-dom';
+import { NotFound } from '../../pages';
+
+type LinkFacadeProps = LinkProps & {
+  // add any other props you want to pass to the Link component
+};
+
+type RouteConfig = {
+  path: string;
+  element: React.ReactNode;
+  // add any other route props here
+};
+
+type RouterFacadeProps = {
+  routes: RouteConfig[];
+  // add any other props that the underlying Routes component needs here
+};
+export const AppRouter = (props: RouterFacadeProps) => {
+  const { routes } = props;
+  const routeConfigs = routes.filter((route) => route.element !== null);
+
+  return (
+    <Router>
+      <Routes>
+        {routeConfigs.map((route, index) => (
+          <Route
+            key={index}
+            path={route.path}
+            element={route.element as React.ReactNode}
+          ></Route>
+        ))}
+        <Route path="*" element={<NotFound />} />
+      </Routes>
+    </Router>
+  );
+};
```

Comment:
```suggestion
export const RouterFacade = ({ routes, ...props }: RouterFacadeProps) => {
  return (
    <Router>
      <Routes {...props}>
        {routes.map(({ path, element }) => (
          <Route key={path} path={path} element={element} />
        ))}
        <Route path="*" element={<NotFound />} />
      </Routes>
    </Router>
  );
};
```

Reply:
I suggested that because a) you didn't have to coerce the type on line 35 because we already know the shape from the type declared earlier.
b) destructuring the properties directly makes the code neat. 

What you do you think?

Reply:
Great Suggestion @shyakadavis .

Reply:
I was thinking that to do like that there is no  harm effects will come up with; 
but to remove them also nothing harm will come with; Consequently I am going to do as you suggested! 
Thank you for your valuable review @shyakadavis  and @mucyoalexaxel